src/app-layer-parser.c

```diff
@@ -130,9 +130,15 @@ struct AppLayerParserState_ {
 
     /* State version, incremented for each update.  Can wrap around. */
     uint8_t version;
+
+    /* Indicates which loggers that have already logged the transaction,
+       using the logger id (power of two) as a bitwise flag. */
+    uint32_t logged;
```

Comment:
I think the way to go is a variant on @jasonish https://github.com/inliniac/suricata/pull/1992/commits/4de928f61287978e79ae4bc57c2ee96c691dc94d patch. The patch is boolean, in our use case we're looking at storing a uint16_t or uin32_t per tx. Not sure if we'd need per direction flags. Not if the logger id's are unique I think.